illumos-utils/src/link.rs

```diff
@@ -155,6 +155,7 @@ pub struct InvalidLinkKind(String);
 /// Note that the "ownership" of the VNIC is based on convention;
 /// another process in the global zone could also modify / destroy
 /// the VNIC while this object is alive.
+#[derive(Clone)]
```

Comment:
This structure has a `drop` implementation that would get invoked repeatedly if we clone this object. Why is it safe/necessary to clone this?